.github/workflows/multi_build.yml

```diff
@@ -6,23 +6,58 @@ on:
   pull_request:
 
 jobs:
+  setup-tools:
+    runs-on: "ubuntu-20.04"
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.21"
+          check-latest: true
+      - name: Cache Tools
+        id: tool-cache
+        uses: actions/cache@v3
+        with:
+          path: /home/runner/go/bin
+          key: tooling-${{ runner.os }}-${{ hashFiles('./Makefile') }}
```

Comment:
This should be based on `internals/tools/go.sum`, as opposed to the Makefile, otherwise old tool versions may be cached in CI.